tests/kafkatest/tests/connect/connect_plugins_rest_test.py

```diff
@@ -0,0 +1,250 @@
+        assert 1 == configs['error_count']
+        actual_config_values = self.get_config_values(configs)
+        self.validate_config_value(expected_config_values, actual_config_values)
+
+        # Test the case that all the configs are provided
+        props = {
+            'name': 'test',
+            'connector.class': self.FILE_SOURCE_CONNECTOR,
+            'tasks.max': 1,
+            'topic': 'file-topic',
+            'file': 'file.txt'
+        }
+
+        expected_config_values = {
+            'name': {'name': 'name', 'value': 'test', 'errors': [], 'recommended_values': [], 'visible': True},
+            'connector.class': {'name': 'connector.class', 'value': self.FILE_SOURCE_CONNECTOR, 'errors': [], 'recommended_values': [], 'visible': True},
+            'tasks.max': {'name': 'tasks.max', 'value': 1, 'errors': [], 'recommended_values': [], 'visible': True},
+            'topics': {'name': 'topics', 'value': [], 'errors': [], 'recommended_values': [], 'visible': True},
+            'file': {'name': 'file', 'value': 'file.txt', 'errors': [], 'recommended_values': [], 'visible': True},
+            'topic': {'name': 'topic', 'value': 'file-topic', 'errors': [], 'recommended_values': [], 'visible': True}
+        }
+        configs = self.validate_config_def(self.FILE_SOURCE_CONNECTOR, self.FILE_SOURCE_ALL_CONFIGS, props)
+        # Should have zero errors
+        assert 0 == configs['error_count']
+        actual_config_values = self.get_config_values(configs)
+        self.validate_config_value(expected_config_values, actual_config_values)
+
+        # Test the case that all the configs are provided
+        props = {
+            'name': 'test',
+            'connector.class': self.FILE_SINK_CONNECTOR,
+            'tasks.max': 1,
+            'topics': 'file-topic',
+            'file': 'file-sink.txt'
+        }
+
+        expected_config_values = {
+            'name': {'name': 'name', 'value': 'test', 'errors': [], 'recommended_values': [], 'visible': True},
+            'connector.class': {'name': 'connector.class', 'value': self.FILE_SINK_CONNECTOR, 'errors': [], 'recommended_values': [], 'visible': True},
+            'tasks.max': {'name': 'tasks.max', 'value': 1, 'errors': [], 'recommended_values': [], 'visible': True},
+            'topics': {'name': 'topics', 'value': ['file-topic'], 'errors': [], 'recommended_values': [], 'visible': True},
+            'file': {'name': 'file', 'value': 'file-sink.txt', 'errors': [], 'recommended_values': [], 'visible': True},
+        }
+        configs = self.validate_config_def(self.FILE_SINK_CONNECTOR, self.FILE_SINK_ALL_CONFIGS, props)
+        # Should have zero errors
+        assert 0 == configs['error_count']
+        actual_config_values = self.get_config_values(configs)
+        self.validate_config_value(expected_config_values, actual_config_values)
+
+    def validate_plugins(self, expected, actual):
+        assert expected == actual
```

Comment:
No need for this method since it's just an assert. I had to scroll all the way down here just to figure out if it did something different...